fairroot/base/sim/FairModule.cxx

```diff
@@ -210,55 +218,50 @@ void FairModule::ProcessNodes(TList* aList)
         vList = new FairVolumeList();
     }
 
-    TListIter iter(aList);
-    FairGeoNode* node = nullptr;
-    FairGeoNode* MotherNode = nullptr;
-    FairVolume* volume = nullptr;
-    FairRuntimeDb* rtdb = FairRun::Instance()->GetRuntimeDb();
-    FairGeoParSet* par = static_cast<FairGeoParSet*>(rtdb->getContainer("FairGeoParSet"));
-    TObjArray* fNodes = par->GetGeoNodes();
-    while ((node = static_cast<FairGeoNode*>(iter.Next()))) {
-
+    auto rtdb = FairRun::Instance()->GetRuntimeDb();
+    auto par = static_cast<FairGeoParSet*>(rtdb->getContainer("FairGeoParSet"));
+    auto parNodes = par->GetGeoNodes();
+    for (auto node : TRangeDynCast<FairGeoNode>(nodes)) {
+        if (!node) {
+            continue;
+        }
         node->calcLabTransform();
-        MotherNode = node->getMotherNode();
-        volume = new FairVolume(node->getTruncName(), fNbOfVolumes++);
+
+        auto volume = std::make_unique<FairVolume>(node->getTruncName(), fNbOfVolumes);
         volume->setRealName(node->GetName());
-        vList->addVolume(volume);
-        volume->setGeoNode(node);
-        volume->setCopyNo(node->getCopyNo());
 
-        if (MotherNode != 0) {
-            volume->setMotherId(node->getMCid());
-            volume->setMotherCopyNo(MotherNode->getCopyNo());
+        auto addedVol = vList->addVolume(std::move(volume));
+        if (!addedVol) {
+            continue;
+        }
```

Comment:
I know I proposed this code! So my fault for mentioning this now…

Could we have a `LOG(warn)` here? AFAICT this code runs only during "initialization" and the user might want to know if some volumes just get dropped due to "duplicate checks".